### PR TITLE
feat(xresources): xresources parser and queries

### PR DESCRIPTION
**Pull request by ValdezFOmar**
Parser for `.Xresources` or `.Xdefaults` files.

### CODE REVIEW

File: queries/xresources/folds.scm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+  (comment)+
+  (elif_directive)
+  (elifdef_directive)
+  (else_directive)
+  (if_directive)
+  (ifdef_directive)
+  (include_directive)+
+  (preprocessor_comment)
```
Comment (ribru17) [inline queries/xresources/folds.scm:9]: Would you mind moving the quantified captures to the end of this alternation? You shouldn't have to, but I've noticed a weird but in tree-sitter where folds of different node kinds in the alternations are merged if they are a quantified one followed by a regular one...

File: queries/xresources/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  "#elifndef"
+  "#else"
+  "#endif"
+  (directive)
+] @keyword.directive
+
+(identifier) @variable
+
+(define_directive
+  name: (identifier) @constant)
```
Comment (ribru17) [inline queries/xresources/highlights.scm:28]: Maybe `@constant` should be `@constant.macro`?

**Review by ribru17 — APPROVED** [review]

File: queries/xresources/highlights.scm (hunk 2 above)
Comment (ValdezFOmar) [reply queries/xresources/highlights.scm:28]: I thought the same but the C queries use `@constant` for this and I wanted to keep them consistent with those (unless the C queries are in the wrong here).

Comment (ribru17) [reply queries/xresources/highlights.scm:28]: Oh thanks for checking that! Yeah IMO that seems wrong for the C queries to be doing that...

Comment (ValdezFOmar) [reply queries/xresources/highlights.scm:28]: I'll change the captures here and open another PR changing the captures for the C queries if that is the desired behavior.

**Pull request merged**